src/applications/vaos/services/appointment/transformers.v2.js

```diff
@@ -206,10 +210,19 @@ export function transformVAOSAppointment(appt) {
         'YYYY-MM-DDTHH:mm:ss',
       )}.999`,
     }));
-    const hasReasonCode = appt.reasonCode?.coding?.length > 0;
+    const hasReasonCode =
+      getAppointmentInfoFromComments(appt.reasonCode?.text, 'reasonCode')
+        .length > 0 || appt.reasonCode?.coding?.length > 0;
+    const reasonCode =
+      getAppointmentInfoFromComments(appt.reasonCode?.text, 'reasonCode')
+        .length > 0
+        ? getAppointmentInfoFromComments(appt.reasonCode?.text, 'reasonCode')[0]
+        : appt.reasonCode?.coding?.[0];
```

Comment:
Any chance this could be refactored? The multiple calls to 'getAppointmentInfFromComments' seem like it could be reduced to one call.

Reply:
@vbahinwillit I have refactored 'getAppointmentInfFromComments' calls in transformer.